src/Request.php

```diff
@@ -145,34 +163,13 @@ public function getPHPReleaseVersion() {
 	private function readVersion($versionString, array $server) {
 		$version = explode('x', $versionString);
 
-		if (count($version) === 9 || count($version) === 12) {
-			$this->majorVersion = (int)$version['0'];
-			$this->minorVersion = (int)$version['1'];
-			$this->maintenanceVersion = (int)$version['2'];
-			$this->revisionVersion = (int)$version['3'];
-			$this->installationMtime = (float)$version['4'];
-			$this->lastCheck = (int)$version['5'];
-			$this->channel = $version['6'];
-			$this->edition = $version['7'];
-			$this->build = explode(' ', substr(urldecode($version['8']), 0, strrpos(urldecode($version['8']), ' ')))[0];
-			$this->remoteAddress = (isset($server['REMOTE_ADDR']) ? $server['REMOTE_ADDR'] : '');
-
-			# TODO remove this once there are no Nextcloud 11.0 betas out there
-			if ($this->majorVersion === 11 &&
-				$this->minorVersion === 0 &&
-				$this->maintenanceVersion === 0 &&
-				in_array($this->revisionVersion, array(2, 4, 5)) &&
-				$this->channel === ''
-			) {
-				$this->channel = 'stable';
-			}
-
-			// starting with nextcloud 17 there is no production channel anymore
-			if ($this->channel === 'production') {
-				$this->channel = 'stable';
-			}
-
-			if(count($version) === 12) {
+		$this->remoteAddress = (isset($server['REMOTE_ADDR']) ? $server['REMOTE_ADDR'] : '');
+		switch (count($version)) {
+			case 14:
+				$this->isSubscriber = (int) $version[13];
```

Comment:
does it mean we rely on the client to provide reliable subscription information?

Reply:
Yes,I implemented it here : https://github.com/nextcloud/server/pull/39264
I didn't find any other way, but we can discuss it if you have some time :)